firebase-perf/src/main/java/com/google/firebase/perf/transport/RateLimiter.java

```diff
@@ -104,6 +104,22 @@ private boolean isDeviceAllowedToSendNetworkEvents() {
     return samplingBucketId < validNetworkSamplingBucketIdThreshold;
   }
 
+  /**
+   * Returns whether device is allowed to send Fragment screen trace events based on Fragment screen
+   * trace sampling rate.
+   */
+  private boolean isDeviceAllowedToSendFragmentScreenTraces() {
+    float validFragmentSamplingBucketIdThreshold = configResolver.getFragmentSamplingRate();
+    return samplingBucketId < validFragmentSamplingBucketIdThreshold;
```

Comment:
Is this the same samplingBucketId used for traces? If so, what are the pros and cons in using a different sampling bucketId? I was biased towards creating a new sampling bucketId for fragment traces. But checking if there are any negates around that.

Reply:
Good point. I used the same bucketId because Network and Trace shared the same one as well. But after thinking about this I think we need to change to different bucketId like you said. 

I don't think it makes a big difference in the case of Network and Trace because they are independent, but since fragment is dependent there's a big problem with using the same bucketId.

#### Suppose we use the same bucketId, and suppose trace sampling rate is 40% and fragment sampling rate is 80%.
Then 40% of all devices will be sampled to send traces. However, within that 40%, all devices will send fragment traces, because 0.8 > 0.4, and we are using the same bucketId, so rolling the dice only once. If the dice already landed < 0.4, then it must also be < 0.8. 

```
0    0.1    0.2    0.3    0.4    0.5    0.6    0.7    0.8    0.9    1
             *             |
             *                                         |
```

#### In the independent case, like network and trace, suppose trace sampling rate is 40% and network is 80%.
```
0    0.1    0.2    0.3    0.4    0.5    0.6    0.7    0.8    0.9    1
|         both             |
|         network                                      |
                                                       |     none   |
```

Reply:
Thanks for the exploration Leo. Like we discussed offline, we should go with two separate bucket ids for traces and fragment screen traces. This would enable us to avoid the situation where fragment screen trace sampling flag becomes unusable.